### PR TITLE
mortality template cleaning

### PR DESCRIPTION
**Pull request by aspina7**
starting to go through mortality template and add explanations

### CODE REVIEW

**Comment by aspina7** [comment]
@lukric @zkamvar - in my latest commit I added some messy code to get the column name for which date occured first to get a cause of observation start variable (e.g. arrival, birth, was already there). 
Could one of you look at trying to make it nicer/dplyr please?

**Comment by aspina7** [comment]
@lukric @zkamvar - we need to be using logit method to calculate the confidence intervals do make them not negative. But cant seem to get it to work, can one of you try please?

File: R/tabulate_survey.R [hunk 1]
```diff
@@ -41,24 +45,33 @@ tabulate_survey <- function(x, var, strata = NULL, pretty = TRUE, digits = 1) {
...
-                                                                 proportion = TRUE,
-                                                                 var = "ci",
-                                                                 na.rm = TRUE)
-                                 )
+  # loop over the values
+  for (i in p) { 
+    print(i)
+    if (y[y[[1]] == i, ]$n > 0) {
+      x <- srvyr::mutate(x, this = i)
+      res[[i]] <- survey::svyciprop(~I(dummy == this), x, method = "logit")
```
Comment (aspina7) [inline R/tabulate_survey.R]: not confident enough to start digging in all of this and change stuff ... but here, if you use srvyr::survey_mean and specify proportion= TRUE and prop_method, then it will use survey::svyciprop. 
Also the reason you got glm.fit error may be because of zero counts in a grop (log reg not converging) - maybe we need to consider one of the others. From reading on ze interwebz it seems like svyciprop has a method called "xlogit" which matches the CDC method of calculating CIs for surveys. This however wouldnt address our zeros issue... would probably be the beta or likelihood methods - to run a poisson or negative binomial type regression...

Comment (zkamvar) [reply R/tabulate_survey.R]: I already [tried setting `proportion = TRUE`](https://github.com/R4EPI/sitrep/pull/98/commits/11638a0d528599818e122b7bd3b95ed9373aabaf#diff-5a50b278625ae42546a5708309ee2d8cR55), but it failed

The zero counts are exactly what's throwing everything off. As svyciprop states in the documenation: 

 > All methods undercover for probabilities close enough to zero or one, but "beta", "likelihood", "logit", and "logit" are noticeably better than the other two. None of the methods will work when the observed proportion is exactly 0 or 1.

I think for zero counts, we will simply have to report NA proportions.

**Comment by zkamvar** [comment]
Allright, I have corrected the CI for the survey proportions. It currently uses the logit method and it's possible to change to xlogit if needed. The only thing I noticed was that the xlogit didn't seem to work without stratification, and I'm not sure why. @aspina7 @lukric, can you take a look to make sure things are cromulent?

**Comment by aspina7** [comment]
note to self: 
* [ ] need to update gen_data to set household arrival equal to born date (if not na) and same for leave and death (if not na) - also add example of logical checks for this to the browsing section of the report

File: inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd [hunk 2]
```diff
@@ -516,24 +516,46 @@ descriptive(study_data_cleaned,
 Compare the proportions in each age group between your sample and the source population.
```
Comment (aspina7) [inline inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: @zkamvar - added in fishers exact test to compare for each age group if the proportion sampled differed significantly from the actual population. 
Not sure this is the most appropriate approach - I know you can do a regular chisq test on the whole table, but dont feel that gives enough information on where the bias is (can infer from the residuals, but thats messy); also can be thrown off by one cell being unexpected...
Also remember we had discussions about correcting for multiple testing... 
Any better ideas?
Also code needs massively simplifying! (i think? - or can we leave as is with more explanation?)

Comment (zkamvar) [reply inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: I believe we can use a binomial test for this and then correct for multiple comparisons with `p.adjust()`

File: inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd [hunk 3]
```diff
@@ -463,7 +464,11 @@ study_data_cleaned <- study_data_cleaned %>%
                 period_end   = "recall_end",
                 datecol      = "enddate",
                 datereason   = "endcause" 
-               )
+               ) %>%
+  # label those that were present at the start/end
+  mutate(startcause = if_else(startdate == recall_start, "Present at start", startcause)) %>%
```
Comment (aspina7) [inline inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: ah didnt realise you meant in the script - coouldve just done this myself rather than bother you!! sorry

File: inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd [hunk 4]
```diff
@@ -605,6 +608,50 @@ descriptive(study_data_cleaned,
   kable(digits = 2)
 ```
 
+```{r visualise_population_flow}
```
Comment (aspina7) [inline inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: @zkamvar when you have time can you try and work your magic on this please? its not agreeing with me completely...

Comment (aspina7) [reply inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: would be nice i think to have the text either side of the boxes maybe ? 
Got it from the [fundamentals of data visualisation book](https://github.com/clauswilke/dataviz/blob/master/nested_proportions.Rmd)

**Comment by aspina7** [comment]
merging for now to work on vaccination templates

File: inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd [hunk 5]
```diff
@@ -252,7 +253,7 @@ unique(study_data_raw$q4_q6_sex)
 #  function)
 study_data_raw %>% 
   filter(q137_q35_died_date < q88_q33_born_date) %>% 
-  select(matches("fact_0_id"))
+  select("fact_0_id")
```
Comment (zkamvar) [inline inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: Why did you remove the matches here?

File: inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd (hunk 4 above)
Comment (zkamvar) [reply inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: Unfortunately, having the text on either side of the diagram (instead of centered in the bars) is a lot of work :sweat:

File: inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd (hunk 5 above)
Comment (aspina7) [reply inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: just not sure it adds much, and extra layer which might confuse people... ? 
(Really the names should just be fixed..... but thats a different story)

Comment (zkamvar) [reply inst/rmarkdown/templates/mortality/skeleton/skeleton.Rmd]: If the "matches" isn't there, then the script throws an error unless the column exists. That's why I put it there.
